hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -176,28 +177,34 @@ public void notifyException(Throwable exception) {
             return;
         }
 
-        if ((isBindToSingleConnection() && exception instanceof IOException)
-                || System.currentTimeMillis() > retryTimeoutPointInMillis) {
+        if (isBindToSingleConnection() && exception instanceof IOException) {
             clientInvocationFuture.complete(exception);
             return;
         }
 
-        if (isRetrySafeException(exception)
+        boolean isAllowedToRetry = isRetrySafeException(exception)
                 || invocationService.isRedoOperation()
-                || (exception instanceof TargetDisconnectedException && clientMessage.isRetryable())) {
-            try {
-                ClientExecutionServiceImpl executionServiceImpl = (ClientExecutionServiceImpl) this.executionService;
-                executionServiceImpl.schedule(this, RETRY_WAIT_TIME_IN_SECONDS, TimeUnit.SECONDS);
-            } catch (RejectedExecutionException e) {
-                if (logger.isFinestEnabled()) {
-                    logger.finest("Retry could not be scheduled ", e);
-                }
-                clientInvocationFuture.complete(exception);
+                || (exception instanceof TargetDisconnectedException && clientMessage.isRetryable());
+        if (!isAllowedToRetry) {
+            clientInvocationFuture.complete(exception);
+            return;
+        }
+
+        long msAfterTimeout = System.currentTimeMillis() - retryTimeoutPointInMillis;
```

Comment:
can we rename 'retryTimeoutPointInMillis' to 'retryExpirationMillis' 

I would rename 'msAfterTimeout' to something like 'remainingMillis'. If it is a negative value, it has expired.